### DISCUSSION TITLE
Merge face with compound into one entity

### DISCUSSION THREAD

**Discussion by Kristoff6** [CadQuery/cadquery#1433] [Q&A] [answered]
How can a face 'F' be merged/added/united/fused/... with a compound 'C' to make one entity, a compound I think? 'F' and 'C' lay in the same plane and overlap.

I've tried:

- C.fuse(F)
- C.fuse(F, glue=True) # hoped this would do the trick, but unfortunately not
- cq.Compound.makeCompound([C, F])
- S = cq.Shell.makeShell([F])
  S.fuse(C, glue=True) 

When exported as a STEP file, all of those create multiple parts for F and C separately, not one.

The background is that FEA boundary conditions need to be applied only on the areas 'C', not on the whole face 'F'. So if its multiple parts, they desintegrate during simulation.
Alternatively joining them in FEA with additional BC's can be done, but is a lot of work. So I'm hoping on a simple fix in CQ.

Some hints are much appreciated.

**Comment by Kristoff6** [comment]
An update: `fuse` apparently links the positional attributes but doesn't create one shape as I think to understand from the documentation.

After many hours, I still did not find a CQ solution. So I tried solving it with additional BC’s in FEA, but unfortunately that’s not successful either at this point.

It would be so much more easy if the 2 components/compounds/shapes can be combined into one.

Alternatively, placing a wire onto a face, starting and stopping at an edge, might also work. I've tried that, but not succesful. Up to now that affects the shape of the face. Something like `face = face.addWire(wire)` or `face = face.addInternalWire(wire)` would be great.

Any suggestions / insights?

**Reply by jmwright** [reply]
You could look at the fused assembly export in CadQuery. It works on faces and not just on assembly objects. `AddShape` and `AddSubShape` may be of interest to you. https://github.com/CadQuery/cadquery/blob/7bc8c00dc1a81633daea4810e04762d61a2abd5a/cadquery/occ_impl/assembly.py#L418

**Reply by Kristoff6** [reply, marked as answer]
Thank you @jmwright for, again, giving inspiration!

Yesterday I finally, found something that looks ok in CAD and could work: `combined = cq.Shell.makeShell([C, F])`
Just now the FEA simulation finished and confirmed it. Finally, case closed!
